addons/popochiu/editor/main_dock/tab_audio/tab_audio.gd

```diff
@@ -0,0 +1,353 @@
+# value.
+var _audio_files_to_assign := {}
+var _audio_cues_to_create := []
+var _created_audio_cues := 0
+
+@onready var _asp: AudioStreamPlayer = $AudioStreamPlayer
+@onready var _btn_scan_files: Button = %BtnScanAudioFiles
+@onready var unassigned_group: PopochiuGroup = %UnassignedGroup
+@onready var _groups := {
+	PopochiuResources.AudioTypes.MUSIC : {
+		array = "mx_cues",
+		group = %MusicGroup
+	},
+	PopochiuResources.AudioTypes.SOUND_EFFECT : {
+		array = "sfx_cues",
+		group = %SFXGroup
+	},
+	PopochiuResources.AudioTypes.VOICE : {
+		array = "vo_cues",
+		group = %VoiceGroup
+	},
+	PopochiuResources.AudioTypes.UI : {
+		array = "ui_cues",
+		group = %UIGroup
+	}
+}
+
+
+#region Godot ######################################################################################
+func _ready() -> void:
+	$PopochiuFilter.groups = _groups
+	_btn_scan_files.icon = get_theme_icon("Search", "EditorIcons")
+	
+	# Connect to childrens signals
+	_btn_scan_files.pressed.connect(search_audio_files)
+	
+	# Connect to helpers signals
+	PopochiuEditorHelper.signal_bus.audio_cues_deleted.connect(delete_rows)
+	
+	# Connect to Editor signals
+	EditorInterface.get_resource_filesystem().sources_changed.connect(_on_sources_changed)
+
+
+#endregion
+
+#region Public #####################################################################################
+func fill_data() -> void:
+	# Look for audio files without AudioCue
+	_put_audio_cues_in_group()
+	search_audio_files()
+
+
+func search_audio_files() -> void:
+	var fs_directory: EditorFileSystemDirectory =\
+	EditorInterface.get_resource_filesystem().get_filesystem_path(SEARCH_PATH)
+	
+	if not fs_directory:
+		return
+	
+	_read_directory(fs_directory)
+	
+	if _audio_cues_to_create.is_empty():
+		return
+	
+	var progress_dialog := await PopochiuEditorHelper.show_progress()
+	progress_dialog.label.text = "Creating PopochiuAudioCues..."
+	progress_dialog.progress_bar.max_value = _audio_cues_to_create.size()
+	
+	EditorInterface.get_resource_filesystem().sources_changed.disconnect(_on_sources_changed)
+	_created_audio_cues = 0
+	for arr in _audio_cues_to_create:
+		await _create_audio_cue(arr[0], arr[1])
+		
+		progress_dialog.progress_bar.value = _created_audio_cues
+		await get_tree().create_timer(0.1).timeout
+	
+	_put_audio_cues_in_group()
+	_audio_cues_to_create.clear()
+	
+	EditorInterface.get_resource_filesystem().sources_changed.connect(_on_sources_changed)
+	progress_dialog.close()
+
+
+func delete_rows(filepaths: Array) -> void:
+	for filepath in filepaths:
+		if filepath in _audio_files_to_assign:
+			_audio_files_to_assign[filepath].free()
+			_audio_files_to_assign.erase(filepath)
+		elif filepath in _audio_files_in_group:
+			_audio_files_in_group[filepath].free()
+			_audio_files_in_group.erase(filepath)
+
+
+#endregion
+
+#region Private ####################################################################################
+func _on_sources_changed(exist: bool) -> void:
+	if Engine.is_editor_hint():
```

Comment:
Why this check?

Reply:
Because you want the plugin to check this only while working in the Editor. If I'm not wrong, the Output was trhowing some errors when using the Debugger when the game was running. But I'll double check and put a comment if needed.

Reply:
Done.